egs/lre07/v2/run.sh

```diff
@@ -0,0 +1,238 @@
+utils/fix_data_dir.sh data/lre07
+
+src_list="data/sre08_train_10sec_female \
+    data/sre08_train_10sec_male data/sre08_train_3conv_female \
+    data/sre08_train_3conv_male data/sre08_train_8conv_female \
+    data/sre08_train_8conv_male data/sre08_train_short2_male \
+    data/sre08_train_short2_female data/ldc96* data/lid05d1 \
+    data/lid05e1 data/lid96d1 data/lid96e1 data/lre03 \
+    data/lre07 data/lre09"
+# Remove any spk2gender files that we have: since not all data
+# sources have this info, it will cause problems with combine_data.sh
+for d in $src_list; do rm -f $d/spk2gender 2>/dev/null; done
+
+utils/combine_data.sh data/train_unsplit $src_list
+
+# original utt2lang will remain in data/train_unsplit/.backup/utt2lang.
+utils/apply_map.pl -f 2 --permissive local/lang_map.txt  < data/train_unsplit/utt2lang  2>/dev/null > foo
+cp foo data/train_unsplit/utt2lang
+rm foo
+
+local/split_long_utts.sh --max-utt-len 120 data/train_unsplit data/train
+
+echo "**Language count in i-Vector extractor training (after splitting long utterances):**"
+awk '{print $2}' data/train/utt2lang | sort | uniq -c | sort -nr
+
+# This commented script is an alternative to the above utterance
+# splitting method. Here we split the utterance based on the number of 
+# frames which are voiced, rather than the total number of frames.
+# max_voiced=3000 
+# local/vad_split_utts.sh --max-voiced $max_voiced data/train_unsplit $mfccdir data/train
+
+
+# We have skipped the VTLN warping step below unlike the v1 recipe of lre07
+use_vtln=false
+if $use_vtln; then
+  for t in train lre07; do
+    cp -r data/${t} data/${t}_novtln
+    rm -r data/${t}_novtln/{split,.backup,spk2warp} 2>/dev/null || true
+    steps/make_mfcc.sh --mfcc-config conf/mfcc_vtln.conf --nj 12 --cmd "$train_cmd" \
+       data/${t}_novtln exp/make_mfcc $mfccdir 
+    lid/compute_vad_decision.sh data/${t}_novtln exp/make_mfcc $mfccdir
+  done
+
+  # Vtln-related things:
+  # We'll use a subset of utterances to train the GMM we'll use for VTLN
+  # warping.
+  utils/subset_data_dir.sh data/train_novtln 5000 data/train_5k_novtln
+
+  # Note, we're using the speaker-id version of the train_diag_ubm.sh script, which
+  # uses double-delta instead of SDC features to train a 256-Gaussian UBM.
+  sid/train_diag_ubm.sh --nj 12 --cmd "$train_cmd" data/train_5k_novtln 256 \
+    exp/diag_ubm_vtln
+  lid/train_lvtln_model.sh --mfcc-config conf/mfcc_vtln.conf --nj 12 --cmd "$train_cmd" \
+     data/train_5k_novtln exp/diag_ubm_vtln exp/vtln
+
+  for t in lre07 train; do
+    lid/get_vtln_warps.sh --nj 12 --cmd "$train_cmd" \
+       data/${t}_novtln exp/vtln exp/${t}_warps
+    cp exp/${t}_warps/utt2warp data/$t/
+  done
+fi
+
+
+
+#Uncomment the following part if use_vtln=true above
+#utils/fix_data_dir.sh data/train
+#utils/filter_scp.pl data/train/utt2warp data/train/utt2spk > data/train/utt2spk_tmp
+#cp data/train/utt2spk_tmp data/train/utt2spk
+#utils/fix_data_dir.sh data/train
```

Comment:
Instead of commenting out this block, can we just put it in an if statement? E.g.,
`if $use_vtln; then
  utils/fix_data_dir.sh data/train
  utils/filter_scp.pl data/train/utt2warp data/train/utt2spk > data/train/utt2spk_tmp
  cp data/train/utt2spk_tmp data/train/utt2spk
  utils/fix_data_dir.sh data/train
fi`